### PR TITLE
Implement Inspection for std::set, std::unordered_set and the like

### PR DESCRIPTION
**Pull request by markuspf**

### CODE REVIEW

File: lib/Inspection/include/Inspection/Access.h [hunk 1]
```diff
@@ -60,6 +60,8 @@ template<class Inspector, class T>
     return f.tuple(x);
   } else if constexpr (detail::IsMapLike<TT>::value) {
     return f.map(x);
+  } else if constexpr (detail::IsSetLike<TT>::value) {
+    return f.set(x);
```
Comment (markuspf) [inline lib/Inspection/include/Inspection/Access.h:64]: Might be able to use list here now

File: lib/Inspection/include/Inspection/LoadInspectorBase.h [hunk 2]
```diff
@@ -413,6 +421,21 @@ struct LoadInspectorBase : InspectorBase<Derived, Context> {
...
+  template<class T>
+  Status processSet(T& set) {
+    std::size_t idx = 0;
+    return this->self().doProcessList([&](auto value) -> Status {
+      auto ff = this->self().make(value);
+      typename T::value_type val;
+      if (auto res = process(ff, val); !res.ok()) {
+        return {std::move(res), std::to_string(idx), Status::ArrayTag{}};
+      }
+      set.insert(std::move(val));
```
Comment (markuspf) [inline lib/Inspection/include/Inspection/LoadInspectorBase.h:433]: I couldn't find a neater way to process a set, beacuse  the function we found with `IsSetLike` has to be used after all.

Comment (mpoeter) [reply lib/Inspection/include/Inspection/LoadInspectorBase.h:433]: I think we can ditch this whole function and handle everything in `processList` like this:
```
  template<class T>
  Status processList(T& list) {
    constexpr isSet = detail::IsSetLike<T>::value;
    std::size_t idx = 0;
    return this->self().doProcessList([&](auto value) -> Status {
      auto ff = this->self().make(value);
      typename T::value_type val;
      if (auto res = process(ff, val); !res.ok()) {
        return {std::move(res), std::to_string(idx), Status::ArrayTag{}};
      }
      if constexpr (isSet) {
        list.insert(std::move(val));
      } else {
        list.push_back(std::move(val));
      }
      ++idx;
      return {};
    });
  }
  ```

File: lib/Inspection/include/Inspection/SaveInspectorBase.h [hunk 3]
```diff
@@ -78,6 +78,14 @@ struct SaveInspectorBase : InspectorBase<Derived, Context> {
...
   }
 
+  template<class T>
+  [[nodiscard]] auto set(T const& set) {
+    auto& f = this->self();
+    return f.beginArray()                                                     //
+           | [&]() { return f.processList(std::begin(set), std::end(set)); }  //
+           | [&]() { return f.endArray(); };                                  //
+  }
+
```
Comment (mpoeter) [inline lib/Inspection/include/Inspection/SaveInspectorBase.h:88]: IMHO this should not be needed anymore.

**Review by mpoeter — APPROVED** [review]

File: lib/Inspection/include/Inspection/SaveInspectorBase.h (hunk 3 above)
Comment (markuspf) [reply lib/Inspection/include/Inspection/SaveInspectorBase.h]: Indeed.

**Pull request merged**